src/main/scala/is/hail/annotations/aggregators/RegionValueSumAggregator.scala

```diff
@@ -0,0 +1,144 @@
+package is.hail.annotations.aggregators
+
+import is.hail.asm4s._
+import is.hail.expr._
+import is.hail.expr.ir._
+import is.hail.annotations._
+import is.hail.expr.RegionValueAggregator
+
+import scala.reflect.ClassTag
+
+object RegionValueSumAggregator {
+  private def seqOp[Agg >: Null : ClassTag : TypeInfo, T : ClassTag](t: Type): (Code[RegionValueAggregator], Code[_], Code[Boolean]) => Code[Unit] =
```

Comment:
`SeqOp[Agg >: RegionValueAggregator, ...` is more specific?

Reply:
The bound would be `Agg >: Null <: RegionValueAggregator`, I'll add this in the `ApplyAggOp` PR.

Reply:
Wait, what?  Agg extends RegionValueAggregator, right?  And U >: T if U extends T, right?  So Agg >: RegionValueAggregator >: Null, that last of which is implied by the RegionValueAggregator bound?  Or am I being stupid here?

Reply:
AFAIK, `>:` is read as "is a super type of", so `Agg` should be a subtype of, `<:` RegionValueAggregator (cause it `extends` it) and should be a super type of `Null` (which was a hack I used to indicate that something was a Reference type, I have forgotten why `AnyRef` wasn't the right choice ...)

Reply:
Yeah, here's a Scala page about it https://docs.scala-lang.org/tour/lower-type-bounds.html

Reply:
Honestly, I think we should change the bounds on `CodeObject` to be subtype of `AnyRef`, this `Null` thing is kind of silly.

I think the issue was that `T <: AnyRef` isn't sufficient to write `val t: T = null`, but that's not necessary to call a method.

Reply:
It builds just fine for me with just `Agg <: RegionValueAggregator`.

You need it to be a subtype of `AnyRef` for the checkcast to work and you get that for free from `RegionValueAggregator`.  I do admit I'm not sure how the `Null` constraint on `CodeObject` is getting satisfied in this case.

I still advocate adding a region value aggregator bound.

Reply:
Yeah RegionValueAggregator bound makes the most sense. I'll get rid of Null in the AggOp PR.